build/target-repository/docs/rector_rules_overview.md

```diff
@@ -2961,6 +2961,29 @@ Remove `@return` docblock with same type as defined in PHP
 
 <br>
 
+### RemoveUselessTypeInParamTagRector
+
+Remove the type of the `@param` docblock if the parameter type is defined in the function signature
+
+- class: [`Rector\DeadCode\Rector\ClassMethod\RemoveUselessTypeInParamTagRector`](../rules/DeadCode/Rector/ClassMethod/RemoveUselessTypeInParamTagRector.php)
+
+```diff
+ class SomeClass
+ {
+     /**
+-     * @param string $a
+-     * @param string $b description
++     * @param $a
++     * @param $b description
```

Comment:
this is wrong, and param tag become mixed, and there is already `RemoveUselessParamTagRector` to remove useless param tag https://github.com/rectorphp/rector-src/blob/main/build/target-repository/docs/rector_rules_overview.md#removeuselessparamtagrector for it

Reply:
The param doesn't become mixed ; the type is optional, as stated [here](https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/param.html). It can be omitted and avoid duplicating info. I still think this could be a great addition. The goal here is not to remove useless param, it removes useless type in `@param` when it is already present in the method signature 🙂

Reply:
you can create your own custom rule for that.